include/sys/zfs_context.h

```diff
@@ -766,4 +766,7 @@ extern void spl_fstrans_unmark(fstrans_cookie_t);
 extern int spl_fstrans_check(void);
 
 #endif /* _KERNEL */
+
+#define	ZFS_ARRAY_SIZE(a)	(sizeof (a) / sizeof (*a))
```

Comment:
The Linux kernel provides an `ARRAY_SIZE` macro for this but there is no equivalent version defined in user space.  We ran in to a similar issue in the extended iostats patch and opted there to handle it by defining `ARRAY_SIZE` in `lib/libspl/include/sys/sysmacros.h`.  It looks like illumos also defines ARRAY_SIZE in `sys/sysmacros.h` so that's shouldn't cause any portability troubles.